subprojects/gdk_rust/gdk_electrum/src/interface.rs

```diff
@@ -0,0 +1,946 @@
+                        .into_derivation_path()?;
+                    info!(
+                        "input#{} prev_output:{:?} derivation_path:{:?}",
+                        i, prev_output, derivation_path
+                    );
+
+                    let (script_sig, witness) =
+                        self.internal_sign_bitcoin(&tx, i, &derivation_path, out.value);
+
+                    out_tx.input[i].script_sig = script_sig;
+                    out_tx.input[i].witness = witness;
+                }
+                let tx = BETransaction::Bitcoin(out_tx);
+                info!(
+                    "transaction final size is {} bytes and {} vbytes",
+                    tx.serialize().len(),
+                    tx.get_weight() / 4
+                );
+                info!("FINALTX inputs:{} outputs:{}", tx.input_len(), tx.output_len());
+                tx.into()
+            }
+            NetworkId::Elements(_) => {
+                let mut tx: elements::Transaction =
+                    elements::encode::deserialize(&hex::decode(&request.hex)?)?;
+                self.blind_tx(&mut tx)?;
+
+                for i in 0..tx.input.len() {
+                    let prev_output = tx.input[i].previous_output;
+                    info!("input#{} prev_output:{:?}", i, prev_output);
+                    let prev_tx = self
+                        .db
+                        .get_liquid_tx(&prev_output.txid)?
+                        .ok_or_else(|| Error::Generic("cannot find tx in db".into()))?;
+                    let out = prev_tx.output[prev_output.vout as usize].clone();
+                    let derivation_path = self
+                        .db
+                        .get_path(&out.script_pubkey)?
+                        .ok_or_else(|| Error::Generic("can't find derivation path".into()))?
+                        .into_derivation_path()?;
+
+                    let (script_sig, witness) =
+                        self.internal_sign_elements(&tx, i, &derivation_path, out.value);
+
+                    tx.input[i].script_sig = script_sig;
+                    tx.input[i].witness.script_witness = witness;
+                }
+
+                let fee: u64 =
+                    tx.output.iter().filter(|o| o.is_fee()).map(|o| o.minimum_value()).sum();
+                let tx = BETransaction::Elements(tx);
+                info!(
+                    "transaction final size is {} bytes and {} vbytes and fee is {}",
+                    tx.serialize().len(),
+                    tx.get_weight() / 4,
+                    fee
+                );
+                info!("FINALTX inputs:{} outputs:{}", tx.input_len(), tx.output_len());
+                tx.into()
+            }
+        };
+
+        let changes_used = request.changes_used.unwrap_or(0);
+        if changes_used > 0 {
+            info!("tx used {} changes", changes_used);
+            self.db.increment_index(Index::Internal, changes_used)?;
```

Comment:
The problem of increasing it when creating it is that `create_transaction` is called multiple times by the apps so that you can easily run off the gap limit.

At the moment there is no way from the GUI to put aside the tx and sign it later, however, it could become a problem if we allow this in the future. In that case, we probably have to save the indexes at export time